codespell fix typos in the main cvmfs/ codebase, add codespell workflow to catch any new typos

yarikoptic opened the pull request:
- [x] workflow was moved to a separate PR
- [x] ~~one typo fix is "functional": `quarantaine/` folder is being renamed to `quarantine/`.  May be should be reverted?~~ reverted, thus no more pertinent.
- there are some typos still left in test cases happen someone wants to practice codespell'-ing ;)

yarikoptic committed:
rudimentary codespellrc

yarikoptic committed:
Manually fixing some typos which were ambigous

yarikoptic committed:
more skips for codespell

yarikoptic commented:
the longer PR stays unattended, the more conflicts it would encounter, the more effort for typos-free-glorious-cvmfs would need to be invested.  Please advise on either I should bother addressing the conflicts.

jblomer commented on cvmfs/publish/cmd_zpipe.cc:
```diff
@@ -28,7 +28,7 @@
 
 #include "duplex_zlib.h"
 
-#if defined(MSDOS) || defined(OS2) || defined(WIN32) || defined(__CYGWIN__)
```
Needs to be reverted

jblomer commented on cvmfs/swissknife_zpipe.cc:
```diff
@@ -26,7 +26,7 @@
 #include "duplex_zlib.h"
 #include "swissknife_zpipe.h"
 
-#if defined(MSDOS) || defined(OS2) || defined(WIN32) || defined(__CYGWIN__)
```
Needs to be reverted

yarikoptic replied:
sorry about that, good job spotting! I will just exclude `msdos` entirely.

jblomer reviewed (changes requested):
Many thanks for the contribution!

The rename of the folder indeed needs to be reverted.

Also, can the workflow please be taken out as a separate PR.

yarikoptic committed:
ignore quarantaine

yarikoptic committed:
Ignore msdos since used in directives, codespell manages to screw up

yarikoptic committed:
[DATALAD RUNCMD] Run codespell throughout fixing some typos

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "codespell -w cvmfs",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^

yarikoptic committed:
[DATALAD RUNCMD] Run codespell throughout some selected test files with easy fixes

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "codespell -w test/run.sh test/common/migration_tests/common.sh test/micro-benchmarks/bm_util.h test/src/017-dnstimeout/main test/src/019-httptimeout/main test/unittests/t_fetch.cc test/unittests/t_reflog.cc test/src/705-evict/main",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^

yarikoptic committed:
no need to spellcheck .rtf files (found in packaging)

yarikoptic committed:
ignore one packaging file since has false positives

yarikoptic committed:
[DATALAD RUNCMD] Run codespell throughout packaging

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "codespell -w packaging",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^

yarikoptic commented:
done, ready for re-review

jblomer commented:
@cernvm-bot cpplint

jblomer commented:
@cernvm-bot tidy

jblomer commented:
@cernvm-bot unittest

jblomer commented:
@cernvm-bot cloudtest

jblomer commented:
@yarikoptic in principle looks good to me, after addressing the linter errors this is good to go.

yarikoptic committed:
Fix a few "Line too long" lint issues

python2 cpplint.py  is now quiet on those files

yarikoptic commented:
FWIW: lint issues were addressed (I even had `python2` locally to ensure that ;))

jblomer commented:
@cernvm-bot unittest
